webapi/src/test/scala/org/knora/webapi/slice/admin/domain/model/UserSpec.scala

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright © 2021 - 2024 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.slice.admin.domain.model
+
+import zio.test.*
+
+import dsp.valueobjects.IriErrorMessages
+
+object UserSpec extends ZIOSpecDefault {
+  private val validUserIri            = "http://rdfh.ch/users/jDEEitJESRi3pDaDjjQ1WQ"
+  private val userIriWithUUIDVersion3 = "http://rdfh.ch/users/cCmdcpn2MO211YYOplR1hQ"
+  private val invalidIri              = "Invalid IRI"
+  private val validPassword           = "pass-word"
+  private val validGivenName          = "John"
+  private val validFamilyName         = "Rambo"
+  private val pwStrength              = PasswordStrength.unsafeMake(12)
+
+  private val userSuite = suite("User")()
+
+  private val usernameSuite = suite("Username")(
+    test("Username must not be empty") {
+      assertTrue(Username.from("") == Left(UserErrorMessages.UsernameMissing))
```

Comment:
When testing please do not reuse production values for the assertion and rather add some redundancy to the test.

For example if someone changes the `UsernameMissing` automatically the test will pass even if the change was unintentional. 

```suggestion
      assertTrue(Username.from("") == Left("Username cannot be empty."))
```

Please also check the other cases below  ↓

Reply:
I think I copied this pattern over from the existing tests... but happy to improve it